### PR TITLE
feat: surplus modal

### PR DESCRIPTION
**Pull request by fairlighteth**
# Summary

- Styles the surplus modal from @alfetopito 
- Has the Tweet button functionality
- On click of the Tweet button fires the track event `onTweetShare`

Screenshot 2023-06-30 at 15 21 24
Screenshot 2023-06-30 at 15 21 36

## Todo:
- Review mobile/responsive
- Review Tweet text: We can specify if they 'saved' or got 'surplus'

### CODE REVIEW

**Review by shoom3301 — APPROVED** [review]
Looks nice :)

File: src/legacy/components/TransactionConfirmationModal/index.tsx [hunk 1]
```diff
@@ -81,7 +81,7 @@ export function TransactionConfirmationModal({
 
 function getWidth(hash: string | undefined, activityDerivedState: ActivityDerivedState | null): number {
   if (activityDerivedState && getActivityState(activityDerivedState) === 'filled') {
-    return 400
+    return 470
```
Comment (alfetopito) [inline src/legacy/components/TransactionConfirmationModal/index.tsx:84]: In that case you can unify the return with line `89`

Change the if condition to:

`if((activityDerivedState && getActivityState(activityDerivedState) === 'filled') || !hash)`

then remove lines 86, 88 and 89, returning `850` in the default case

**Review by alfetopito — APPROVED** [review]
Looks amazing!

One small suggestion regarding the modal size logic

File: src/legacy/components/TransactionConfirmationModal/index.tsx (hunk 1 above)
Comment (fairlighteth) [reply src/legacy/components/TransactionConfirmationModal/index.tsx:84]: Or, can we specify the 470 value specifically for this surplus modal? Not sure that's a specific 'activity'.

File: src/common/pure/TransactionSubmittedContent/SurplusModal.tsx [hunk 2]
```diff
@@ -10,6 +20,128 @@ import { TokenAmount } from 'common/pure/TokenAmount'
 const SELL_SURPLUS_WORD = 'got'
 const BUY_SURPLUS_WORD = 'saved'
 
+function getTwitterText(surplusAmount: string, surplusToken: string) {
+  const surplus = `${surplusAmount} ${surplusToken}`
+  return encodeURIComponent(
+    `Hey, I just ${SELL_SURPLUS_WORD} ${surplus} on @CoWSwap! 🐮💸\n\nStart swapping on swap.cow.fi`
```
Comment (anxolin) [inline src/common/pure/TransactionSubmittedContent/SurplusModal.tsx:26]: shouldn't this message receive the `order.kind` as a parameter (or directly the `surplusWord`)? so it works for buy orders too?

**Comment by fairlighteth** [comment]
Added updates for dark mode:
Screenshot 2023-06-30 at 15 40 27

File: src/common/pure/TransactionSubmittedContent/SurplusModal.tsx (hunk 2 above)
Comment (fairlighteth) [reply src/common/pure/TransactionSubmittedContent/SurplusModal.tsx:26]: Yes you are right. Just committed a fix.

**Comment by elena-zh** [comment]
Hey @fairlighteth , great!
Some issues that I see: 
1. Twitter button looks a bit different to me in comparison with the screenshot
button diff
2. Twitter post: maybe better to add 'extra' into the text? 
extra
3. Twitter: again, for a buy order maybe better to say 'saved extra' ?
saved
4. modal, responsive view: for a tablet resolution the content does not cover the whole screen: 
surface

**Comment by fairlighteth** [comment]
@elena-zh Thank you for reviewing. The first and last point should be fixed.

**Comment by elena-zh** [comment]
@fairlighteth , cases 1, 4 LGTM!

**Comment by fairlighteth** [comment]
@elena-zh I will merge and we can see about point 2 and 3 as this PR aims to focus around the styling aspects.

**Comment by alfetopito** [comment]
> @elena-zh I will merge and we can see about point 2 and 3 as this PR aims to focus around the styling aspects.

@fairlighteth I can take care of that if you want

**Comment by elena-zh** [comment]
> > @elena-zh I will merge and we can see about point 2 and 3 as this PR aims to focus around the styling aspects.
> 
> @fairlighteth I can take care of that if you want

@fairlighteth , @alfetopito , I have created #2780 task for this

**Pull request merged**